06_puppy/mohankrishna/map_store.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+type Puppy struct {
+	ID     uint32
+	Breed  string
+	Colour string
+	Value  int
+}
+
+type Storer interface {
+	CreatePuppy(puppy *Puppy)
+	ReadPuppy(ID uint32) *Puppy
+	UpdatePuppy(ID uint32, pet *Puppy)
+	DeletePuppy(ID uint32) bool
+}
+
+type MapStore struct {
+	m map[uint32]Puppy
+}
+
+func CreateStore() *MapStore {
+	return &MapStore{make(map[uint32]Puppy)}
+}
+
+func (ms *MapStore) CreatePuppy(puppy *Puppy) {
+	ms.m[puppy.ID] = *puppy
```

Comment:
Also, typically Create is responsible for assigning new IDs - user of this package will probably no know which IDs are still free.